sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/BinaryDataSerializer.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.jackson;
+
+import com.azure.core.implementation.util.BinaryDataContent;
+import com.azure.core.implementation.util.BinaryDataHelper;
+import com.azure.core.implementation.util.ByteArrayContent;
+import com.azure.core.implementation.util.FileContent;
+import com.azure.core.implementation.util.FluxByteBufferContent;
+import com.azure.core.implementation.util.InputStreamContent;
+import com.azure.core.implementation.util.SerializableContent;
+import com.azure.core.implementation.util.StringContent;
+import com.azure.core.util.BinaryData;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import java.io.IOException;
+
+/**
+ * Custom serializer for serializing {@link BinaryData}.
+ */
+final class BinaryDataSerializer extends JsonSerializer<BinaryData> {
+    /**
+     * Gets a module wrapping this serializer as an adapter for the Jackson ObjectMapper.
+     *
+     * @return a simple module to be plugged onto Jackson ObjectMapper.
+     */
+    public static SimpleModule getModule() {
+        SimpleModule module = new SimpleModule();
+        module.addSerializer(BinaryData.class, new BinaryDataSerializer());
+        return module;
+    }
+
+    @Override
+    public void serialize(BinaryData value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+        if (value == null) {
+            return;
+        }
+
+        BinaryDataContent content = BinaryDataHelper.getContent(value);
+        if (content instanceof ByteArrayContent || content instanceof FileContent
+            || content instanceof FluxByteBufferContent || content instanceof InputStreamContent) {
+            gen.writeBinary(content.toBytes());
```

Comment:
nit: would it make sense to do it by default (e.g. in case new content is added)

```
if (content instanceof SerializableContent) {
    gen.writeRawValue(content.toString());
} else if (content instanceof StringContent) {
    gen.writeString(content.toString());
} else {
   gen.writeBinary(content.toBytes());
}
```

or just add else clause and log error if content type is unknown?

Reply:
I'll update this to throw an exception as we need explicit support for all content types